src/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/AbstractRequestScope.cs

```diff
@@ -12,11 +12,21 @@ internal abstract class AbstractRequestScope(string name) : IDisposable
 {
     public string Name { get; } = name;
 
+    public string? Language { get; private set; }
+
     public abstract void RecordCancellation();
     public abstract void RecordException(Exception exception);
     public abstract void RecordWarning(string message);
 
     public abstract void RecordExecutionStart();
 
+    /// <summary>
+    /// Sets the language for the request once it has been determined
```

Comment:
this comment coudl be expanded.  i would dive into this maybe not being consistent/apparaent at an earlier point, and the idea of this needing to be figured out serially by the queue to be in a consistent state.